### PR TITLE
Fix #7: support for Scrapinghub Collections

### LINKED ISSUE #7: Add support for Scrapy Cloud Collections
Currently `spider-feeder` can load files from `s3` or local file system.
It would be nice to extend it to load data from [Scrapy Cloud Collections](https://doc.scrapinghub.com/api/collections.html) through https://python-scrapinghub.readthedocs.io/en/latest/client/overview.html#collections.

### PR DESCRIPTION
**Pull request by ejulio**
Fix #7 
I had to add an abstraction for `Store`, this way I can load from files of APIs.
I changed `SPIDERFEEDER_FILE_URI` to `SPIDERFEEDER_INPUT_URI`.
Also, `SPIDERFEEDER_STORES` was added to allow custom backend stores.

What do you think of `colections://` scheme? Is there a better name?

### CODE REVIEW

**Comment by andersonberg** [comment]
I'd say `hubstore` would be a better naming

File: README.md [hunk 1]
```diff
@@ -94,15 +98,36 @@ The interface is class with `__init__(settings: scrapy.Settings)` and `parse(fd:
 ```
 # settings.py
 SPIDERFEEDER_FILE_PARSERS = {
-    's3': 'myproject.my_custom_parser.CustomParser'
+    's3': 'myproject.my_custom_parser.parse'
```
Comment (victor-torres) [inline README.md]: Why you're moving to functions here? I kinda liked the class-based parsers. You can add related code and auxiliar private methods to the class keeping the logic a little bit more organized and even inherit from other parsers.

File: requirements.txt [hunk 2]
```diff
@@ -3,6 +3,9 @@ scrapy
 # for s3:// schemes
 botocore
 
+# for scrapinghub collection
+scrapinghub
```
Comment (victor-torres) [inline requirements.txt]: Not sure why you're keeping a requirements.txt and also specifying requirements in setup.py. It's always better to keep a single source of truth.

Comment (ejulio) [reply requirements.txt]: requirements.txt for development and setup.py for pip install

File: README.md (hunk 1 above)
Comment (ejulio) [reply README.md]: Just realized that a plain function with 2 parameters would be enough.
But I can go back in this change..

**Comment by croqaz** [comment]
My 2 cents: not sure if Hubstorage is the best naming.
Here: https://doc.scrapinghub.com/api/collections.html and Here: https://doc.scrapinghub.com/api/frontier.html and Here: https://doc.scrapinghub.com/api/logs.html
They all start with https://storage.scrapinghub.com/ so they are all about storage.
The Collections seem to be exactly that, ScrapyCloud Collections IHMO.
Don't want to debate the naming too much though, it's just a name.

File: README.md (hunk 1 above)
Comment (victor-torres) [reply README.md]: It's up to you. We can always move back later if we need to.

**Comment by ejulio** [comment]
Given @croqaz comment, I'll rename it to `collections` as it is the path after `storage.scrapinghub.com`.
Then, if we add support for frontier, it can follow the same idea and be set as `hcf`.

**Pull request merged**